packages/platform/api/.github/workflows/fusionauth-api-tests.yml

```diff
@@ -0,0 +1,124 @@
+name: FusionAuth API Tests
+
+on:
+  push:
+    branches: [ main, master, development ]
+  pull_request:
+    branches: [ main, master, development ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    services:
+      # Setup PostgreSQL for Prisma
+      postgres:
+        image: postgres:13
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: testdb
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+      
+      # Setup FusionAuth
+      fusionauth:
+        image: fusionauth/fusionauth-app:latest
+        env:
+          DATABASE_URL: jdbc:postgresql://postgres:5432/fusionauth
+          DATABASE_ROOT_USERNAME: postgres
+          DATABASE_ROOT_PASSWORD: postgres
+          DATABASE_USERNAME: postgres
+          DATABASE_PASSWORD: postgres
+          FUSIONAUTH_APP_MEMORY: 512M
+          SEARCH_TYPE: database
+        ports:
+          - 9011:9011
+
+    strategy:
+      matrix:
+        node-version: [18.x]
+
+    steps:
+    - uses: actions/checkout@v3
+    
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'yarn'
+    
+    - name: Install dependencies
+      run: yarn install --frozen-lockfile
+    
+    - name: Wait for FusionAuth
+      run: |
+        echo "Waiting for FusionAuth to start up..."
+        timeout 300 bash -c 'until curl -s http://localhost:9011/api/status -o /dev/null; do sleep 5; done'
+        echo "FusionAuth is running!"
+    
```

Comment:
this waiting can also go into the docker compose, also add the complete nestjs server in the docker compose and add `FusionAuth` and `postgres` to be the dependencies for that container.

You'll need to create the `Dockerfile` for the Nest app.